packages/core/src/component.test.ts

```diff
@@ -0,0 +1,311 @@
+
+            expect(subscribeWithSelector).toHaveBeenCalledTimes(0)
+            expect(store.getState).toHaveBeenCalledTimes(0)
+            expect(component.bindActionCreators).toHaveBeenCalledTimes(1)
+        })
+        it("calls callRender and connected in connectedCallback", () => {
+            class TestComponent extends Component {
+                callRender = jest.fn()
+                connected = jest.fn()
+            }
+            const component = new TestComponent()
+            component.connectedCallback()
+
+            expect(component.callRender).toHaveBeenCalledTimes(1)
+            expect(component.connected).toHaveBeenCalledTimes(1)
+        })
+        it("does not call unsubscribe if not set", () => {
+            class TestComponent extends Component {
+                disconnected = jest.fn()
+            }
+            const component = new TestComponent()
+            component.disconnectedCallback()
+
+            expect(component.disconnected).toHaveBeenCalledTimes(1)
+        })
+        it("does calls unsubscribe if is set", () => {
+            const store = createEmptyStore(null)
+            class TestComponent extends Component {
+                disconnected = jest.fn()
+                unsubscribe = jest.fn(store.subscribe(null))
+            }
+            const component = new TestComponent()
+            component.disconnectedCallback()
+
+            expect(component.disconnected).toHaveBeenCalledTimes(1)
+            expect(component.unsubscribe).toHaveBeenCalledTimes(1)
+        })
+        it("does not call attributeChangedCallback when values are undefined or equal", () => {
+            class TestComponent extends Component {
+                propChanged = jest.fn()
+
+                attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+                    super.attributeChangedCallback(name, oldValue, newValue)
+                }
+            }
+            const component = new TestComponent()
+            const value = "value1"
+
+            component.attributeChangedCallback("name1", undefined, undefined)
+            component.attributeChangedCallback("name1", undefined, "undefined")
+            component.attributeChangedCallback("name1", "undefined", undefined)
+            component.attributeChangedCallback("name1", value, value)
+
+            expect(component.propChanged).toHaveBeenCalledTimes(0)
+        })
+        it("does not call attributeChangedCallback when values are not undefined & are not equal", () => {
+            class TestComponent extends Component {
+                propChanged = jest.fn()
+
+                attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+                    super.attributeChangedCallback(name, oldValue, newValue)
+                }
+            }
+            const component = new TestComponent()
+
+            const name = "name1", value1 = "value1", value2 = "value2"
```

Comment:
string consts

Reply:
what should i change?

Reply:
The same as above 
https://github.com/one-more/micro-frontends/pull/26#discussion_r348506490

Reply:
all values are already in contsants